packages/webapp/src/components/Filter/FilterMultiSelectV2/index.tsx

```diff
@@ -0,0 +1,251 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { MouseEvent, useEffect, useLayoutEffect, useMemo, useRef } from 'react';
+import Select, {
+  ClearIndicatorProps,
+  components,
+  MenuListProps,
+  MultiValue,
+  MultiValueRemoveProps,
+  OptionProps,
+} from 'react-select';
+import { ComponentFilter, ComponentFilterOption } from '../types';
+import { useState } from 'react';
+import { ReactComponent as XCircle } from '../../../assets/images/x-circle.svg';
+import { useTranslation } from 'react-i18next';
+import TextButton from '../../Form/Button/TextButton';
+import styles from './styles.module.scss';
+import Checkbox from '../../Form/Checkbox';
+import produce from 'immer';
+import { FilterItemProps } from '../FilterGroup';
+import { Label } from '../../Typography';
+
+interface FilterMultiSelectProps extends ComponentFilter {
+  onChange?: FilterItemProps['onChange'];
+  isDisabled?: boolean;
+}
+
+const MultiValueRemove = (props: MultiValueRemoveProps) => (
+  <components.MultiValueRemove {...props}>
+    <XCircle />
+  </components.MultiValueRemove>
+);
+
+const ClearIndicator = (props: ClearIndicatorProps<ComponentFilterOption>) => {
+  const { getValue, clearValue } = props;
+  const value = getValue();
+  const { t } = useTranslation();
+
+  const onClick = (e: MouseEvent) => {
```

Comment:
Hmm, a bit surprised that the stopPropagation + preventDefault aren't enough to keep the filterSelect from re-opening when you hit clear? Ideally it would stay closed in this interaction, right?


https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/dba791ab-ce82-4a8d-8125-c619917df891

Reply:
Ah I thought I had got rid of this 😭 so annoying!

Reply:
So I just need to use the default indicator component as a wrapper for mine, not sure why but it makes sense to use it anyway! Solving this also gets rid of the pills flicker you mentioned above